doc/neps/nep-XXXX-SIMD-optimizations.rst

```diff
@@ -0,0 +1,199 @@
+(name TBD).
+
+
+Diagnostics
+```````````
+- How does a user know what optimizations are enabled for the installed NumPy?
+  Add to `show_config()` output? What about actual runtime capability on the
+  installed system?
+
+
+Workflow for adding a new CPU architecture-specific optimization
+````````````````````````````````````````````````````````````````
+
+NumPy will always have a baseline C implementation for any code that may be
+a candidate for SIMD vectorization.  Now if a contributor wants to add SIMD
+support for some architecture (typically the one of most interest to them),
+this is the proposed workflow:
+
+TODO (see https://github.com/numpy/numpy/pull/13516#issuecomment-558859638,
+needs to be worked out more)
+
+
+
+Backward compatibility
+----------------------
+
+There should be no impact on backwards compatibility.
+
+
+Detailed description
+--------------------
+
+*This section should provide a detailed description of the proposed change.
+It should include examples of how the new functionality would be used,
+intended use-cases and pseudo-code illustrating its use.*
+
+TODO: status today - what instructions are used at build time (SSE2/3 and
+AVX/AVX512 for XXX functionality) and at runtime (some but less, see
+``loops.c.src``)
+
+
+Related Work
+------------
+
+*This section should list relevant and/or similar technologies, possibly in other
+libraries. It does not need to be comprehensive, just list the major examples of
+prior and relevant art.*
+
+- OpenCV
+- PIXMAX
+- Eigen
+- xsimd_
+
+
+Implementation
+--------------
+
+*This section lists the major steps required to implement the NEP.  Where
+possible, it should be noted where one step is dependent on another, and which
+steps may be optionally omitted.  Where it makes sense, each step should
+include a link to related pull requests as the implementation progresses.
+
+Any pull requests or development branches containing work on this NEP should
+be linked to from here.  (A NEP does not need to be implemented in a single
+pull request if it makes sense to implement it in discrete phases).*
+
+Current PRs:
+
+- `gh-13421 improve runtime detection of CPU features <https://github.com/numpy/numpy/pull/13421>`_
+- `gh-13516: enable multi-platform SIMD compiler optimizations <https://github.com/numpy/numpy/pull/13516>`_
+
+**Let's leave description of this out for now. Only do that once the questions
+in the sections above are answered.**
+
+
+Alternatives
+------------
+
+*If there were any alternative solutions to solving the same problem, they should
+be discussed here, along with a justification for the chosen approach.*
+
+A proposed alternative in gh-13516_ is a per CPU architecture implementation of
+SIMD code (e.g., have `loops.avx512.c.src`, `loops.avx2.c.src`, `loops.sse.c.src`,
+`loops.vsx.c.src`, `loops.neon.c.src`, etc.). This is more similar to what
+OpenCV and PIXMAX do. There's a lot of duplication here though, it is likely
```

Comment:
Please remove OpenCV, this is no longer accepted by OpenCV. the code that mentioned in comment  [gh-13516](https://github.com/numpy/numpy/pull/13516#issuecomment-558463897) is part of the legacy code,  OpenCV has different machinizem now similar to what we gonna have, unfortunately, infrastructure isn't documented well and universal intrinsics is not updated yet, we have something called wide universal intrinsics which support AVX2 and AVX512.

Here's an example from the OpenCV source code:
https://github.com/opencv/opencv/blob/master/modules/core/src/arithm.dispatch.cpp
https://github.com/opencv/opencv/blob/master/modules/core/src/arithm.simd.hpp
https://github.com/opencv/opencv/blob/master/modules/core/CMakeLists.txt#L3-L13

Reply:
Interesting, thanks @seiko2plus. Yes, looks like you are correct. I looked at some of your OpenCV PRs, and I do agree it makes the code more readable. Also benchmarking works nicely. Example: https://github.com/opencv/opencv/pull/10406

Reply:
So we can actually use a sentence like: _OpenCV used to have the one-implementation-per-architecture design, but more recently moved to a "universal intrinsics" design that is quite similar to what is proposed in this NEP (example: LINK)._

Reply:
removed OpenCV from "Alternatives and reworked it the "Related Work" section.